src/ui/public/chrome/directives/app_switcher/app_switcher.js

```diff
@@ -1,10 +1,18 @@
 import DomLocationProvider from 'ui/dom_location';
 import { parse } from 'url';
-import { bindKey } from 'lodash';
+import { bindKey, filter } from 'lodash';
 import '../app_switcher/app_switcher.less';
 import uiModules from 'ui/modules';
 import appSwitcherTemplate from './app_switcher.html';
 
+function isNavLinkEnabled(link) {
+  return link.enable === undefined || link.enable;
+}
+
+function isNavLinkShown(link) {
+  return link.show === undefined || link.show;
```

Comment:
I like the idea of defining them explicitly, but from a UI standpoint I still think `disabled` and `hidden` make more sense as property names. There's precedent: `disabled` is a standard HTML attribute for form elements and `visibility: hidden` is used in CSS.